backend/src/main/java/com/google/blackswan/mock/Constant.java

```diff
@@ -16,30 +16,55 @@
 
 import com.google.models.DataInfo;
 import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableList;
+import java.util.Map;
+import java.util.HashMap;
 
 /** Constants used throughout the blackswan mock classes. */
 public final class Constant {
+  // Constants used to filenames. 
+  private static final String FILE_DELIMITER = "-";
+  private static final String FILE_END = "--data.csv";
+  private static final String US_COUNTRY = "us";
+  private static final String JP_COUNTRY = "jp";
+  private static final String UK_COUNTRY = "gb";
+
   // TODO: Create based on external config file (in the future).
-  public static final String INTEREST_US = "Interest Over Time - US";
-  public static final String INTEREST_JP = "Interest Over Time - JP";
-  public static final String INTEREST_UK = "Interest Over Time - UK";
+  public static final String INTEREST = "Interest Over Time - ";
+  public static final String INTEREST_US = INTEREST + US_COUNTRY.toUpperCase();
+  public static final String INTEREST_JP = INTEREST + JP_COUNTRY.toUpperCase();
+  public static final String INTEREST_UK = INTEREST + UK_COUNTRY.toUpperCase();
   public static final String UDON = "Udon";
   public static final String PHO = "Pho";
   public static final String RAMEN = "Ramen";
 
-  // TODO: Build with builder or for-loop.
-  public static final ImmutableMap<DataInfo, String> FILE_LOCATIONS 
-      = ImmutableMap.<DataInfo, String>builder()
-          .put(DataInfo.of(INTEREST_US, UDON), "udon-us--data.csv")
-          .put(DataInfo.of(INTEREST_US, PHO), "pho-us--data.csv")
-          .put(DataInfo.of(INTEREST_US, RAMEN), "ramen-us--data.csv")
-          .put(DataInfo.of(INTEREST_JP, RAMEN), "ramen-jp--data.csv")
-          .put(DataInfo.of(INTEREST_JP, PHO), "pho-jp--data.csv")
-          .put(DataInfo.of(INTEREST_JP, UDON), "udon-jp--data.csv")
-          .put(DataInfo.of(INTEREST_UK, RAMEN), "ramen-gb--data.csv")
-          .put(DataInfo.of(INTEREST_UK, PHO), "pho-gb--data.csv")
-          .put(DataInfo.of(INTEREST_UK, UDON), "udon-gb--data.csv")
-          .build();
+  // Constants used to generate filenames map. 
+  private static final ImmutableList<String> COUNTRIES_LIST = 
+      ImmutableList.<String>builder()
+      .add(US_COUNTRY).add(JP_COUNTRY).add(UK_COUNTRY).build();
+  private static final ImmutableList<String> DIMENSIONS = 
+      ImmutableList.<String>builder()
+      .add(UDON).add(PHO).add(RAMEN).build();
```

Comment:
could you not just do ImmutableList.of(US_COUNTRY, JP_COUNTRY, UK_COUNTRY)?

Reply:
ohh, yes good point. i thought that was meant for making copies, but i think it should work.